kafka/producer/base.py

```diff
@@ -76,6 +77,20 @@ def _send_upstream(queue, client, codec, batch_time, batch_size,
             defaults to True.
         stop_timeout (int or float, optional): number of seconds to continue
             retrying messages after stop_event is set, defaults to 30.
+
+    Events emitted (no args unless specified):
+        async.producer.connect.succeed
+        async.producer.connect.fail
+        async.producer.stop(unsent_messages)
+        async.producer.queue.pop(topic_partition, msg, key)
+        async.producer.request.send(requests)
+        async.producer.request.succeed(request)
+        async.producer.request.error(error_cls, request)
+        async.producer.request.retry(request)
+        async.producer.backoff(time_in_ms)
+        async.producer.metadata.refresh
+        async.producer.metadata.refresh.fail
```

Comment:
I made all this stuff up, I just went through this function and added an event for everything important that it does. And I'm only adding it here because this is all we care about currently.